src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+            _spiDevice = spiDevice;
+            _gpioController = gpioController ?? new GpioController(PinNumberingScheme.Logical);
+            _shouldDispose = shouldDispose;
+            _pinBusy = pinBusy;
+            _pinNss = pinNss;
+            _gpioController.OpenPin(_pinBusy, PinMode.Input);
+            _gpioController.OpenPin(_pinNss, PinMode.Output);
+
+            // Check the version
+            var (product, firmware, eeprom) = GetVersion();
+            if ((product.Major == 0) || (firmware.Major == 0) || (eeprom.Major == 0))
+            {
+                throw new IOException($"Not a valid PN5180");
+            }
+        }
+
+        /// <summary>
+        /// Dispose
+        /// </summary>
+        public void Dispose()
+        {
+            // If the RF field is on, try to switch it off
+            try
+            {
+                SetRadioFrequence(false);
+            }
+            catch (Exception)
+            {
+                // We don't do anything on purpose. Trying to switch off
+            }
+
+            if (_shouldDispose)
+            {
+                _spiDevice?.Dispose();
+                _gpioController?.Dispose();
+            }
+        }
+
+        #region EEPROM
+
+        /// <summary>
+        /// Get the Product, Firmware and EEPROM versions of the PN8150
+        /// </summary>
+        /// <returns>A tuple with the Product, Firmware and EEPROM versions</returns>
+        public (Version product, Version firmware, Version eeprom) GetVersion()
+        {
+            Span<byte> versionAnswer = stackalloc byte[6];
+
+            var ret = ReadEeprom(EepromAddress.ProductVersion, versionAnswer);
+            if (!ret)
+            {
+                return (null, null, null);
+            }
+
+            var product = new Version(versionAnswer[1], versionAnswer[0]);
+            var firmware = new Version(versionAnswer[3], versionAnswer[2]);
+            var eeprom = new Version(versionAnswer[5], versionAnswer[4]);
+            return (product, firmware, eeprom);
+        }
+
+        /// <summary>
+        /// Get the PN5180 identifier, this is a 16 byte long
+        /// </summary>
+        /// <param name="identifier">A 16 byte buffer</param>
+        /// <returns>True if success</returns>
+        public bool GetIdentifier(Span<byte> identifier)
+        {
+            if (identifier.Length != 16)
+            {
+                throw new ArgumentException($"Identifier must be 16 bytes long");
+            }
+
+            return ReadEeprom(EepromAddress.DieIdentifier, identifier);
+        }
+
+        /// <summary>
+        /// Read the full EEPROM
+        /// </summary>
+        /// <param name="eeprom">At least 255 bytes buffer</param>
+        /// <returns>True if success</returns>
+        public bool ReadAllEeprom(Span<byte> eeprom)
+        {
+            if (eeprom.Length > 255)
```

Comment:
the description says `at least 255`, this says `less than 255`

Reply:
Also since you used the word ReadAll I'm fine with `eeprom.Length != 255`

Reply:
yes, good catch, will adjust both the description and the condition